src/main/java/net/glowstone/inventory/crafting/CraftingManager.java

```diff
@@ -478,40 +480,55 @@ private void loadRecipes() {
             Map<String, Map<String, Object>> ingredients = (Map<String, Map<String, Object>>) data
                     .get("ingredients"); // NON-NLS
             for (Entry<String, Map<String, Object>> entry : ingredients.entrySet()) {
+                entry.getValue().put("v",1); //Not legacy.
+
                 ItemStack stack = ItemStack.deserialize(entry.getValue());
-                recipe.setIngredient(entry.getKey().charAt(0), stack.getData());
+                recipe.setIngredient(entry.getKey().charAt(0), stack);
             }
 
             shapedRecipes.add(recipe);
         }
 
         // shapeless
         for (Map<?, ?> data : config.getMapList("shapeless")) { // NON-NLS
+
+            ((Map<String, Object>) data.get("result")).put("v", 1); //Non-legacy. (1.13)
+
             ItemStack resultStack =
                     ItemStack.deserialize((Map<String, Object>) data.get("result")); // NON-NLS
+
             NamespacedKey key = readKey(data, resultStack);
             ShapelessRecipe recipe = new ShapelessRecipe(key, resultStack);
 
             List<Map<String, Object>> ingreds
                     = (List<Map<String, Object>>) data.get("ingredients"); // NON-NLS
             for (Map<String, Object> entry : ingreds) {
-                recipe.addIngredient(ItemStack.deserialize(entry).getData());
+                entry.put("v", 1); //Non-legacy Materials
+
+                recipe.addIngredient(ItemStack.deserialize(entry));
             }
 
             shapelessRecipes.add(recipe);
         }
 
         // furnace
         for (Map<?, ?> data : config.getMapList("furnace")) { // NON-NLS
+
+            ((Map<String, Object>) data.get("input")).put("v", 1); //Non-legacy.
+            ((Map<String, Object>) data.get("result")).put("v", 1); //Non-legacy.
+
             ItemStack inputStack
-                    = ItemStack.deserialize((Map<String, Object>) data.get("input")); // NON-NLS
+                    = ItemStack.deserialize((Map<String, Object>) data.get("input")); // NON-NLS;
             ItemStack resultStack
-                    = ItemStack.deserialize((Map<String, Object>) data.get("result")); // NON-NLS
+                    = ItemStack.deserialize((Map<String, Object>) data.get("result")); // NON-NLS;
+
             NamespacedKey key = readKey(data, resultStack);
-            float xp = ((Number) data.get("xp")).floatValue(); // NON-NLS
-            furnaceRecipes.add(
-                    new FurnaceRecipe(key, resultStack, inputStack.getType(),
-                            inputStack.getDurability(), xp, 200));
+            //float xp = ((Number) data.get("xp")).floatValue(); // NON-NLS
```

Comment:
Seems like xp was removed from the original recipes.yml. Is that intentional? If not, could you add it back?